spec/sass-spec.coffee

```diff
@@ -144,3 +144,127 @@ describe 'SASS grammar', ->
       expect(tokens[0][1]).toEqual value: ' hi1', scopes: ['source.sass', 'comment.line.sass']
       expect(tokens[1][0]).toEqual value: '  hi2', scopes: ['source.sass', 'comment.line.sass']
       expect(tokens[2][0]).not.toEqual value: 'hi3', scopes: ['source.sass', 'comment.line.sass']
+
+  describe 'correctly tokenizes at-rules and directives with proper syntax', ->
+    it '@function', ->
+      {tokens} = grammar.tokenizeLine '@function function_name($p1, $p2)'
+      expect(tokens[0]).toEqual value: '@', scopes: ['source.sass', 'meta.at-rule.function.sass', 'keyword.control.at-rule.function.sass', 'punctuation.definition.entity.sass']
+      expect(tokens[1]).toEqual value: 'function', scopes: ['source.sass', 'meta.at-rule.function.sass', 'keyword.control.at-rule.function.sass']
+      expect(tokens[3]).toEqual value: 'function_name', scopes: ['source.sass', 'meta.at-rule.function.sass', 'support.function.misc.sass']
+
+    it '@return', ->
+      {tokens} = grammar.tokenizeLine '@return \'border\' + \' \' + \'1px solid pink\''
+      expect(tokens[0]).toEqual value: '@', scopes: ['source.sass', 'meta.at-rule.return.sass', 'keyword.control.return.sass', 'punctuation.definition.entity.sass']
+      expect(tokens[1]).toEqual value: 'return', scopes: ['source.sass', 'meta.at-rule.return.sass', 'keyword.control.return.sass']
+
+    it '@if', ->
+      {tokens} = grammar.tokenizeLine '@if $var == true'
+      expect(tokens[0]).toEqual value: '@', scopes: ['source.sass', 'meta.at-rule.if.sass', 'keyword.control.if.sass', 'punctuation.definition.entity.sass']
+      expect(tokens[1]).toEqual value: 'if', scopes: ['source.sass', 'meta.at-rule.if.sass', 'keyword.control.if.sass']
+
+    it '@else if', ->
+      {tokens} = grammar.tokenizeLine '@else if $var == false'
+      expect(tokens[0]).toEqual value: '@', scopes: ['source.sass', 'meta.at-rule.else.sass', 'keyword.control.else.sass', 'punctuation.definition.entity.sass']
+      expect(tokens[1]).toEqual value: 'else if ', scopes: ['source.sass', 'meta.at-rule.else.sass', 'keyword.control.else.sass']
+
+    it '@while', ->
+      {tokens} = grammar.tokenizeLine '@while 1'
+      expect(tokens[0]).toEqual value: '@', scopes: ['source.sass', 'meta.at-rule.while.sass', 'keyword.control.while.sass', 'punctuation.definition.entity.sass']
+      expect(tokens[1]).toEqual value: 'while', scopes: ['source.sass', 'meta.at-rule.while.sass', 'keyword.control.while.sass']
+
+    it '@for', ->
+      {tokens} = grammar.tokenizeLine '@for $i from 1 through 100'
```

Comment:
Similarly, what about the `from` and the `through`?  Are those tokenized correctly?  Some more examples are below.